config/initializers/logging.rb

```diff
@@ -0,0 +1,40 @@
+require "./app/lib/json_log_formatter"
+
+unless Rails.env.development?
+  Rails.application.configure do
+    ### LOGGING CONFIGURATION ###
+    config.log_level = :info
+
+    # Use JSON log formatter for better support in Splunk. To use conventional
+    # logging use the Logger::Formatter.new.
+    config.log_formatter = JsonLogFormatter.new
+
+    if ENV["RAILS_LOG_TO_STDOUT"].present?
+      config.logger = ActiveSupport::Logger.new($stdout)
+      config.logger.formatter = config.log_formatter
+    end
+
+    # Lograge is used to format the standard HTTP request logging
+    config.lograge.enabled = true
+    config.lograge.formatter = Lograge::Formatters::Json.new
+
+    # Lograge suppresses the default Rails request logging. Set this to true to
+    #  make lograge output it which includes some extra debugging
+    # information.
+    config.lograge.keep_original_rails_log = false
+
+    config.lograge.custom_options = lambda do |event|
+      {}.tap do |h|
+        h[:host] = event.payload[:host]
+        h[:user_email] = event.payload[:user_email]
+        h[:user_organisation_slug] = event.payload[:user_organisation_slug]
+        h[:user_ip] = event.payload[:user_ip]
+        h[:request_id] = event.payload[:request_id]
+        h[:user_id] = event.payload[:user_id]
+        h[:form_id] = event.payload[:form_id] if event.payload[:form_id]
+        h[:page_id] = event.payload[:page_id] if event.payload[:page_id]
+        h[:exception] = event.payload[:exception] if event.payload[:exception]
+      end
```

Comment:
```suggestion
      {}.tap do |h|
        %i[host user_email user_organisation_slug user_ip request_id user_id form_id page_id exception].each do |key|
          h[key] = event.payload[key] if event.payload[key]
        end
      end
```

I was wondering if this is a better way of writing this, we could even declare that list of keys in a const

Reply:
This change makes sense to me, I wonder if we could do it in a separate PR though? I personally think it's better to keep PRs focused on one thing.